### PR TITLE
Call git diff only once

### PR DESCRIPTION
**Pull request by relekang**
* Adds async await
* Changes the flow of the lint-filter to fetch all diff information first and then add send that down the chain to all functions who needs it. This makes it possible to support a much higher number of files if lint warnings in them.
* Move all git utils into its own file

### CODE REVIEW

File: src/utils/git.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const matches = filenameRegex.exec(diff)
+  if (matches === null) {
+    return null
+  }
+  const filename = matches[1]
+  return { filename, ranges: parseDiffRanges(diff) }
+}
+
+export function parseFullDiff(diff) {
+  return _(diff.split('diff --git '))
```
Comment (trygveaa) [inline src/utils/git.js]: You should put a `\n` before diff, so you don't split on occurrences of that in the files.

File: src/utils/git.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  if (!_.isEmpty(matches)) {
+    return matches.map(match => {
+      const [start, end] = /^@@ -\d+,\d+ \+(\d+),(\d+) @@/.exec(match).slice(1, 3)
+      return [parseInt(start, 10), parseInt(start, 10) + parseInt(end, 10)]
+    })
+  }
+  return []
+}
+
+const filenameRegex = /a\/([^\n]+) b\/[^\n]+/
```
Comment (trygveaa) [inline src/utils/git.js]: This should probably have a `^` in the start to only match the one after `diff --git`.

File: src/utils/git.js (hunk 1 above)
Comment (relekang) [inline src/utils/git.js]: That will skip the first one, could always do 
```es6
`\n${diff}`.split('\ndiff --git ')
```

Comment (trygveaa) [inline src/utils/git.js]: Ah, right. Didn't think of that when writing the comment on `filenameRegex` above (it doesn't matter if it skips the first one unless you also add `^` to that).

Then I suggest using a regex, which seems a bit nicer than adding a newline:
```
diff.split(/^diff --git /m)
```

File: src/utils/git.js (hunk 2 above)
Comment (trygveaa) [inline src/utils/git.js]: `.` matches every character _except newline_, so `[^\n]` can be simplified to `.` which is exactly the same.

**Pull request merged**